Adapt to the select_or_other.js updates.

torotil opened the pull request:
The old version also didn’t work with multiple donation amounts. The new version does.

torotil committed:
Adapt to the select_or_other.js updates.

katzmo commented on js/jquery.donation-buttons.js:
```diff
@@ -21,39 +21,25 @@
+
+      // Add class to other text field when other checkbox is checked.
+      $element.bind('select-or-other-update', function (event, data) {
+        $otherText.parent().toggleClass('select-or-other-checked', data.otherSelected);
```
This class is currently not used in any styles and can safely be removed. New assets may use the class on the component wrapper set by select_or_other if needed.

torotil replied:
How about removing it altogether then? The only thing left is the placeholder and that can be just as well set in PHP.

katzmo replied:
Ok, fine for me :)

torotil committed:
Remove rest of $.donationButtons.

- The placeholder wasn’t used and can be implemented in JS.
- We can use the .select-or-other-checked class on the .select-or-other
  for styling.
